src/main/java/coffee/order/domain/food/FoodCategory.java

```diff
@@ -0,0 +1,84 @@
+package coffee.order.domain.food;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static coffee.order.exception.FoodException.FOOD_CATEGORY_GET_NULL_POINTER_EXCEPTION;
+
+public enum FoodCategory {
+
+    COFFEE(
+            new Foods(List.of(
+                    new Food(1L, "에스프레소", 2000, 999),
+                    new Food(2L, "아메리카노", 3000, 999),
+                    new Food(3L, "콜드블루", 4000, 999),
+                    new Food(4L, "카페라떼", 4000, 999)
+            )),
+            "커피",
+            1L
+    ),
+    TEA(
+            new Foods(List.of(
+                    new Food(1L, "그린티", 3000, 999)
+            )),
+            "티",
+            2L
+    ),
+    DESSERT(
+            new Foods(List.of(
+                    new Food(1L, "케익", 6000, 999)
+            )),
+            "디저트",
+            3L
+    );
+
+    private Foods foods;
+    private final String name;
+    private final Long id;
+
+    FoodCategory(Foods foods, String name, Long id) {
+        this.foods = foods;
+        this.name = name;
+        this.id = id;
+    }
+
+    public static Foods findFoodsByCategoryId(Long categoryId) {
+        return findFoodCategoryByCategoryId(categoryId).foods;
+    }
+
+    public static FoodCategory findFoodCategoryByCategoryId(Long categoryId) {
+        return findOptionalFoodCategoryByCategoryId(categoryId)
+                .orElseThrow(FOOD_CATEGORY_GET_NULL_POINTER_EXCEPTION::throwMyException);
+    }
+
+    private static Optional<FoodCategory> findOptionalFoodCategoryByCategoryId(Long categoryId) {
+        return Arrays.stream(FoodCategory.values())
+                .filter(category -> checkSameFoodCategoryId(categoryId, category.id))
+                .findFirst();
+    }
```

Comment:
```suggestion
        public static Foods findFoodsByCategoryId(Long categoryId) {
        return Arrays.stream(FoodCategory.values())
	        .filter(category -> checkSameFoodCategoryId(categoryId, category.id))
	        .map(category -> category.foods)
	        .findFirst()
	        .orElseThrow();
    }
```

크게 메서드가 재활용되는 것 같지 않아 하나로 합쳐도 되지 않을까 생각됩니다.

Reply:
이후에 사용될거 같아서 미리 분리 시켜놓았는데
재사용하지 않으니 오히려 코드가 길어지기만 해버렸네요 !

수정했습니다 !